tests/bluetooth/bsim_bt/bsim_test_eatt/src/main_connect.c

```diff
@@ -88,13 +100,18 @@ BT_CONN_CB_DEFINE(conn_callbacks) = {
 static void test_peripheral_main(void)
 {
 	int err;
-	size_t num_enhanced;
 
 	err = bt_enable(NULL);
 	if (err) {
 		FAIL("Can't enable Bluetooth (err %d)\n", err);
 	}
 
+	struct bt_eatt_cb eatt_cb = {
+		.eatt_chan_connected = eatt_chan_connected,
+	};
```

Comment:
This should probably have been statically allocated, right?

Reply:
Yes, it has to be alive until shutdown.